sdk/storage/azure-storage-blob/azure/storage/blob/_container_client.py

```diff
@@ -1164,113 +1217,92 @@ def set_standard_blob_tier_blobs(
         """
-        access_conditions = get_access_conditions(kwargs.pop('lease', None))
-        if standard_blob_tier is None:
-            raise ValueError("A StandardBlobTier must be specified")
-
-        query_parameters, header_parameters = self._generate_set_tier_options(
-            tier=standard_blob_tier,
-            lease_access_conditions=access_conditions,
-            **kwargs
-        )
-        # To pass kwargs to "_batch_send", we need to remove anything that was
-        # in the Autorest signature for Autorest, otherwise transport will be upset
-        for possible_param in ['timeout', 'lease']:
-            kwargs.pop(possible_param, None)
-
-        reqs = []
-        for blob in blobs:
-            blob_name = _get_blob_name(blob)
-            req = HttpRequest(
-                "PUT",
-                "/{}/{}".format(self.container_name, blob_name),
-                headers=header_parameters
-            )
-            req.format_parameters(query_parameters)
-            reqs.append(req)
+        reqs, options = self._generate_set_tiers_options(standard_blob_tier, *blobs, **kwargs)
 
-        return self._batch_send(*reqs, **kwargs)
+        return self._batch_send(*reqs, **options)
 
     @distributed_trace
     def set_premium_page_blob_tier_blobs(
         self,
-        premium_page_blob_tier,  # type: Union[str, PremiumPageBlobTier]
-        *blobs,  # type: Union[str, BlobProperties]
+        premium_page_blob_tier,  # type: Optional[Union[str, PremiumPageBlobTier]]
+        *blobs,  # type: List[Union[str, BlobProperties, dict]]
         **kwargs
     ):
         # type: (...) -> Iterator[HttpResponse]
-        """Sets the page blob tiers on the blobs. This API is only supported for page blobs on premium accounts.
+        """Sets the page blob tiers on all blobs. This API is only supported for page blobs on premium accounts.
 
         :param premium_page_blob_tier:
             A page blob tier value to set the blob to. The tier correlates to the size of the
             blob and number of allowed IOPS. This is only applicable to page blobs on
             premium storage accounts.
+
+            ..note::
+                If you want to set different tier on different blobs please set this positional parameter to None.
+                Then the blob tier on every BlobProperties will be taken.
+
         :type premium_page_blob_tier: ~azure.storage.blob.PremiumPageBlobTier
-        :param blobs: The blobs with which to interact. This can be a single blob, or multiple values can
+        :param blobs:
+            The blobs with which to interact. This can be a single blob, or multiple values can
             be supplied, where each value is either the name of the blob (str) or BlobProperties.
-        :type blobs: str or ~azure.storage.blob.BlobProperties
+
+            ..note::
+                When then blob type is dict, here's a list of keys you can set:
+                blob name: 'name'
```

Comment:
What do I set here: the name as a string, or BlobProperties....?

Reply:
name as string, should we support both?